<?xml version="1.0"?>
<launch>
  <arg name="output" default="screen"/> <!-- Set output="log" to mute the screen output -->
  <arg name="debug" default="True"/> <!-- Enable debugging information and pictures -->
  <arg name="save" default="False"/> <!-- Enable database dumping in folder logs/ -->
  <arg name="source" default="none"/> <!-- Source of pre-planned data for this experiment -->
  <arg name="name" default="default_experiment"/> <!-- Name of the experiment -->
  <arg name="display" default="action"/> <!-- display mode for baxter {action,debug,face} -->
  <arg name="iterations" default="10000"/> <!-- Number of iterations to run -->
  <arg name="ip_vrpn" default="baxterflowers.local"/> <!-- IP of the VRPN server (like Motive or Arena) -->

  <include file="$(find thr_interaction_controller)/launch/manual.launch" >
    <arg name="scene" value="pobax"/>
    <arg name="display" value="$(arg display)"/>
    <arg name="name" value="$(arg name)"/>
    <arg name="ip" value="$(arg ip_vrpn)"/>
    <arg name="comm_mode" value="ros"/>
  </include>

  <param name="/pobax_playground/perception/debug" value="$(arg debug)" />
  <param name="/pobax_playground/experiment_name" value="$(arg name)" />
  <param name="/pobax_playground/source_name" value="$(arg source)" />
  <param name="/pobax_playground/iterations" value="$(arg iterations)" />
  <param name="/pobax_playground/save" value="$(arg save)" />

  <node name="learning" pkg="pobax_playground" type="learning_node.py" output="$(arg output)" required="true"/>
  <node name="torso" pkg="pobax_playground" type="torso_node.py" output="$(arg output)" respawn="true"/>
  <node name="voice" pkg="pobax_playground" type="voice_node.py" output="$(arg output)" respawn="true"/>
  <node name="perception" pkg="pobax_playground" type="perception_node.py" output="$(arg output)"/>
  <node name="baxter" pkg="pobax_playground" type="baxter_node.py" output="$(arg output)" required="true"/>
  <node name="controller" pkg="pobax_playground" type="controller.py" output="$(arg output)" required="true"/>

</launch>
